web/js/codeworld_shared.js

```diff
@@ -982,41 +1004,39 @@ function saveProjectAsBase(successFunc) {
     const selected = $('#directoryTree').tree('getSelectedNode');
     if (selected) pathToRoot = pathToRootDir(selected);
     if (pathToRoot !== '') {
-        text = `Enter a name for your project in folder <b>${ 
+        text = `Enter a name for your project in folder <b>${
             $('<div>').text(getNearestDirectory()).html().replace(/ /g,
-                '&nbsp;') 
+                '&nbsp;')
         }:`;
     } else {
         text = 'Enter a name for your project:';
     }
 
-    let defaultName;
-    if (window.openProjectName) {
-        defaultName = window.openProjectName;
-    } else {
-        defaultName = '';
-    }
-
     sweetAlert({
         title: Alert.title('Save As', 'mdi-cloud-upload'),
         html: text,
         input: 'text',
-        inputValue: defaultName,
+        inputValue: utils.directoryTree.getCurrentProjectName() || '',
         confirmButtonText: 'Save',
         showCancelButton: true,
-        closeOnConfirm: false
+        closeOnConfirm: false,
     }).then(result => {
         const parent = getNearestDirectory_();
 
         function localSuccessFunc() {
-            const matches = parent.children.filter(n => n.name === result.value && n.type === 'project');
+            const matches = parent.children
+                .filter(n => n.name === result.value && n.type === 'project');
             let node;
+            const type = 'project';
```

Comment:
Just nitpicking, but I don't see how these two constants (`type` and `name`) make anything clearer.

Reply:
Just keeping things DRY. See code right below:

```js
$('#directoryTree').tree(
  'appendNode', {
    id: utils.directoryTree.createNodeId(type, name),
    name,
    type,
    data: JSON.stringify(getCurrentProject()),
  },
  parent
);
```